src/compiler/preprocess/index.ts

```diff
@@ -76,60 +123,107 @@ export default async function preprocess(
 	const filename = (options && options.filename) || preprocessor.filename; // legacy
 	const dependencies = [];
 
-	const preprocessors = Array.isArray(preprocessor) ? preprocessor : [preprocessor];
+	const preprocessors = Array.isArray(preprocessor) ? preprocessor : [preprocessor || {}];
```

Comment:
why the need for the empty object here?

Reply:
preprocessor should default to {} before preprocessor.filename
otherwise we risk "TypeError: Cannot read property filename"

fn preprocess should accept falsy values for preprocessor = noop
i solved this with
```js
	const preprocessors = preprocessor
		? Array.isArray(preprocessor) ? preprocessor : [preprocessor]
		: []; // noop
```